javav2/example_code/acmpca/src/main/java/com/example/acmpca/scenarios/CreatePrivateCertificateAuthorityAD.java

```diff
@@ -0,0 +1,244 @@
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.pkcs.PKCS10CertificationRequest;
+import org.bouncycastle.util.io.pem.PemReader;
+import java.util.Base64; 
+
+import lombok.SneakyThrows;
+
+// snippet-start:[acmpca.CreatePrivateCertificateAuthorityAD.main]
+/**
+ * Before running this Java V2 code example, set up your development 
+ * environment, including your credentials.
+ * 
+ * For more information, see the following documentation topic:
+ * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
+ */
+public class CreatePrivateCertificateAuthorityAD{
+    public static void main(String[] args) throws Exception {
+        final String usage =
+            """
+                Usage: <region> 
+
+                Where:
+                    region - The AWS region (e.g. us-east-1)
+                """;
+
+        if (args.length != 1) {
+          System.out.println(usage);
+          return;
+        }
+
+        String region = args[0];
+
+        // Define custom attributes
+        List<CustomAttribute> customAttributes = Arrays.asList(
+                CustomAttribute.builder()
+                        .objectIdentifier("2.5.4.3") // OID for Common Name
+                        .value("root CA")
+                .build(),
+                CustomAttribute.builder()
+                        .objectIdentifier("0.9.2342.19200300.100.1.25") // OID for Domain Component
+                        .value("example")
+                .build(),
+                CustomAttribute.builder()
+                        .objectIdentifier("0.9.2342.19200300.100.1.25") // OID for Domain Component
+                        .value("com")
+                .build()
+            
+        );
+
+        // Define a CA subject.
+        ASN1Subject subject = ASN1Subject.builder()
+                .customAttributes(customAttributes)
+                .build();
+
+        // Define the CA configuration.
+        CertificateAuthorityConfiguration configCA = 
+            CertificateAuthorityConfiguration.builder()
+                .keyAlgorithm(KeyAlgorithm.EC_PRIME256_V1)
+                .signingAlgorithm(SigningAlgorithm.SHA256_WITHECDSA)
+                .subject(subject)
+                .build();
+
+        // Define a certificate authority type
+        CertificateAuthorityType CAtype = CertificateAuthorityType.ROOT;
+
+        // ** Execute core code samples for Root CA activation in sequence **
+        AcmPcaClient client = ClientBuilder(region);
+        String rootCAArn = CreateCertificateAuthority(configCA, CAtype, client);
+        String csr = GetCertificateAuthorityCsr(rootCAArn, client);
+        String rootCertificateArn = IssueCertificate(rootCAArn, csr, client);
+        String rootCertificate = GetCertificate(rootCertificateArn, rootCAArn, client);
+        ImportCertificateAuthorityCertificate(rootCertificate, rootCAArn, client);
+    }
+
+    private static AcmPcaClient ClientBuilder(String region) {
```

Comment:
I know it's not your name, but `ClientBuilder` is a bad name for this function. Try coming up with something better :)